File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int size = nums.size(), insPos = 0;

        for(int i = 0 ; i < size ; i++) {
            if(nums[i] != 0) {
                nums[insPos] = nums[i];
                insPos++;
            }
        }

        for(int i = insPos ; i < size ; i++)
            nums[i] = 0;

    }
};

File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let size = nums.length;
    let left = 0, right = 1;

    while(right < size) {
        if(nums[left] != 0) left++;
        else if(nums[right] != 0) {
            nums[left] = nums[left] ^ nums[right];
            nums[right] = nums[left] ^ nums[right];
            nums[left] = nums[left] ^ nums[right];

            left++;
        }
        right++;
    }
};